### ISSUE TITLE
Question about rule MD051

### ISSUE THREAD

**Issue by mundodisco8** [DavidAnson/markdownlint#729]
Hi,

I have a question about rule MD051

I generate a TOC using Markdown Enhanced Preview. It follows what looks like what's described in MD051's page as Github heading algorithm.

```MD
# Title

## Changelog

| Date | Author | Changes |
| - | - | - |
| 20/01/2021 | Me | First Version of this doc |
| 24/02/2021 | Someone Else | Changed the abstract |

## Table of Contents
<!-- If using Markdown Enhanced Preview in VS Code, the ToC can be autogenerated. Otherwise, do it by hand :S. Press shift+enter in the next line WITH MARKDOWN ENHANCED PREVIEW TAB open in VS Code (making sure SNIPPET EXECUTION IS ENABLED, see this https://shd101wyy.github.io/markdown-preview-enhanced/#/code-chunk) to autogenerate a new ToC -->

<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=2 orderedList=false} -->

<!-- code_chunk_output -->

- [Title](#-title-)
  - [Changelog](#-changelog-)
  - [Table of Contents](#-table-of-contents-)
  - [Introduction](#-introduction-)
  - [Intended Audience](#-intended-audience-)
  - [Content](#-content-)

<!-- /code_chunk_output -->

## Introduction

## Intended Audience

## Content
```

In the TOC, spaces have been replaced with `-`. Markdownlint complains that those fragments are not valid...

Removing the heading and trailing dashes, (for example changing `- [Changelog](#-changelog-)` to `- [Changelog](#changelog)`)) stops the warning.

I tried to read MD051 carefully but I don't understand if the error is in the generator or in the rule, sorry, can you clarify that the rule is doing the correct thing?

**Comment by DavidAnson** [comment]
Given the heading `## Intended Audience`, GitHub generates the link `#intended-audience`. That is the behavior this rule is enforcing and it sounds like that's what you're observing.

I am aware that some other syntaxes work with GitHub, and that other platforms use entirely different algorithms. However, this rule seeks to enforce the canonical GitHub syntax which I consider to be the one it generates.

**Comment by mundodisco8** [comment]
I think it's fair to say that the linter doesn't really do anything wrong. I'll ask on the generator side (Markdown Preview Enhanced) and if I don't get a reply there, I think I can live with Rule 51 disabled.

Thanks!
